src/components/Table/BatchActionTable.tsx

```diff
@@ -1,28 +1,35 @@
-import { useRef, useState } from "react";
+import { useEffect, useRef, useState } from 'react';
+import exampleData from './example-data.json';
 
 const BatchActionTable = () => {
   const table = useRef<HTMLTdsTableBodyElement>(null);
   const modal = useRef<HTMLTdsModalElement>(null);
-  const [data, setData] = useState<string>("[]");
+  const [data, setData] = useState<string>('[]');
+
+  useEffect(() => {
+    if (table.current) {
+      table.current.bodyData = exampleData;
+    }
+  });
 
   const handleClick = () => {
     if (table.current) {
-      if (table.current.getAttribute("data-selected-rows")) {
-        setData(table.current.getAttribute("data-selected-rows") as string);
+      if (table.current.getAttribute('data-selected-rows')) {
+        setData(table.current.getAttribute('data-selected-rows') as string);
       }
       modal.current?.showModal();
     }
   };
   return (
     <>
       <tds-modal ref={modal} id="my-modal" size="lg" actions="static">
-        <h5 className="tds-modal-headline" slot="tds-modal-headline">
+        <h5 className="tds-modal-headline" slot="header">
```

Comment:
When selecting the top most Checkbox in the Batch Action Table all the rows are not selected (as I believe is the expected behavior, correct me if that's not the case).
![image](https://github.com/scania-digital-design-system/tegel-react-demo/assets/48509966/df72ced3-abfb-4f63-a0d5-7f6fa5de0324)

Reply:
I have rebased the branch and now the issues are gone.